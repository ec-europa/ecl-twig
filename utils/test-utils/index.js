const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
