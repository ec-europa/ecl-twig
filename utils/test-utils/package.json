{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
