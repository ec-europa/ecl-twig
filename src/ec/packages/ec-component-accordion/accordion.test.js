/* eslint-disable no-param-reassign */

import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

import demoData from './demo/data';

demoData.items.forEach(item => {
  item.toggle.icon.path = 'static/icons';
});

describe('EC - Accordion', () => {
  const template = path.resolve(__dirname, './accordion.html.twig');
  const render = params => renderTwigFileAsNode(template, params);

  test('renders correctly', () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
