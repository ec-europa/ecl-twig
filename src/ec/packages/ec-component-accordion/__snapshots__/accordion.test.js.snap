// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Accordion renders correctly 1`] = `
<dl
  class="ecl-accordion"
  data-ecl-accordion="true"
  role="presentation"
>
  <dt
    aria-level="3"
    role="heading"
  >
    <button
      aria-controls="accordion-example"
      class="ecl-button ecl-button--ghost ecl-accordion__toggle"
      data-ecl-accordion-toggle="true"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <svg
          aria-controls="accordion-example"
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-button__icon ecl-button__icon--before"
          data-ecl-accordion-toggle="true"
          focusable="false"
        >
          <use
            xlink:href="static/icons#general--digital"
          />
        </svg>
        <span
          class="ecl-button__label"
        >
          Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
        </span>
      </span>
    </button>
  </dt>
  <dd
    class="ecl-accordion__content"
    hidden=""
    id="accordion-example"
    role="region"
  >
    The College of Commissioners held today the first weekly meeting of 2019 which was devoted to discussing the challenges of this new year. Commissioners used the opportunity to take stock and discuss the year ahead, including the European elections in May and other important milestones ahead.
  </dd>
  <dt
    aria-level="3"
    role="heading"
  >
    <button
      aria-controls="accordion-example2"
      class="ecl-button ecl-button--ghost ecl-accordion__toggle"
      data-ecl-accordion-toggle="true"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <svg
          aria-controls="accordion-example2"
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-button__icon ecl-button__icon--before"
          data-ecl-accordion-toggle="true"
          focusable="false"
        >
          <use
            xlink:href="static/icons#general--energy"
          />
        </svg>
        <span
          class="ecl-button__label"
        >
          Energy union and climate
        </span>
      </span>
    </button>
  </dt>
  <dd
    class="ecl-accordion__content"
    hidden=""
    id="accordion-example2"
    role="region"
  >
    A balanced and progressive trade policy to harness globalisation
  </dd>
  <dt
    aria-level="3"
    role="heading"
  >
    <button
      aria-controls="accordion-example3"
      class="ecl-button ecl-button--ghost ecl-accordion__toggle"
      data-ecl-accordion-toggle="true"
      type="button"
    >
      <span
        class="ecl-button__container"
      >
        <svg
          aria-controls="accordion-example3"
          aria-hidden="true"
          class="ecl-icon ecl-icon--s ecl-button__icon ecl-button__icon--before"
          data-ecl-accordion-toggle="true"
          focusable="false"
        >
          <use
            xlink:href="static/icons#general--regulation"
          />
        </svg>
        <span
          class="ecl-button__label"
        >
          Delivery of last pending proposals, a common Destiny of unity, the hour of European Democracy
        </span>
      </span>
    </button>
  </dt>
  <dd
    class="ecl-accordion__content"
    hidden=""
    id="accordion-example3"
    role="region"
  >
    In the modern global economy trade is essential for growth, jobs and competiveness, and the EU is committed to maintaining an open and rules-based trading system. With the rising threat of protectionism and weakened commitment of large players to global trade governance, the EU must take the lead.
  </dd>
</dl>
`;
